app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    User.find(user_id).increment!(:posts_counter)
+  end
+
```

Comment:
- I will suggest that you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) -` after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private
- Example:
```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```